backend/app/models/user.rb

```diff
@@ -0,0 +1,22 @@
+class User < ApplicationRecord
+  belongs_to :account, optional: true
+
+  devise :database_authenticatable, :confirmable, :registerable,
+    :recoverable, :rememberable, :validatable
+
+  enum role: {light: 0, investor: 1, project_developer: 2}, _default: :light
```

Comment:
What do you mean? You mean, that instead of `has_one` could be `has_many` that is true from db schema point of view, also account investor/project_developer relation is not polymorphic.

That role here is just some cache. I did it differently before like:

```
 def role
    return nil if account.nil?
    return "investor" if account.investor.present?
    return "project_developer" if account.project_developer.present?

    nil
  end
``` 

I wonder if to use strings instead of numbers for all rails enums (for sake of consistency with not-rails enums), like `light: "light"` etc. What do you think?

Reply:
Ohh, this is completely fine, I just though that you can be investor and project developer at same time, because of
```
has_one :investor
has_one :project_developer
```
so kind of expected that user will have `roles` instead of `role`. Whatever, I am fine with this, no big deal